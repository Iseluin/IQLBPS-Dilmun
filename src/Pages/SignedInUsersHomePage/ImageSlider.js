import React, {useState} from "react" ;

function ImageSlider({ slides }){
    const [currentIndex , setCurrentIndex] = useState(0) ;
    const imageSrc = `url(${slides[currentIndex]})` ;

    const goToPrevious = () => {
        const isFirstSlide = currentIndex === 0 ;
        const newIndex = isFirstSlide ? slides.length-1: currentIndex-1;
        setCurrentIndex(newIndex) ;
    }

    const goToNext = () => {
        const isLastSlide = currentIndex === slides.length-1 ;
        const newIndex = isLastSlide ? 0 :currentIndex+1 ;
        setCurrentIndex(newIndex) ;
    }
    // Now the the image and the url is passed correctly to the ImageSlider component but it's not renderd for some reason 
    // also both functions of go to previous and go to next are working correctly 
     console.log(imageSrc);
     console.log(slides[currentIndex]);

    return(
        <div className="w-full h-full p-10 relative">
            <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" strokeWidth={1.5} stroke="currentColor" className="w-6 h-6 absolute inset-y-0 left-0" onClick={goToNext}>
                <path strokeLinecap="round" strokeLinejoin="round" d="M18.75 19.5l-7.5-7.5 7.5-7.5m-6 15L5.25 12l7.5-7.5" />
            </svg>
            <img src={imageSrc} alt='best selling list'/>
            {/* <img src="https://placeimg.com/800/480/tech/grayscale" alt="trial "/> */}
            <h5 className="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white">Backpack</h5>
                <p className="mb-3 font-normal text-gray-700 dark:text-gray-400">Here are the biggest enterprise technology acquisitions of 2021 so far, in reverse chronological order.</p>
                <button type='submit' className="inline-flex items-center text-background px-3 py-2 rounded bg-secondary ml-2 hover:bg-primary transition duration-300">
                    Read more
                </button>
            <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" strokeWidth={1.5} stroke="currentColor" className="w-6 h-6 absolute inset-y-0 right-0" onClick={goToPrevious}>
                <path strokeLinecap="round" strokeLinejoin="round" d="M11.25 4.5l7.5 7.5-7.5 7.5m-6-15l7.5 7.5-7.5 7.5" />
            </svg>
        </div>
    );
}

export default ImageSlider ;