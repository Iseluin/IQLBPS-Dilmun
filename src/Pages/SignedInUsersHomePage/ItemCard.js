import React from 'react'

const ItemCard = ({ imgSrc }) => {
    return (
        <div className="max-w-sm bg-white border border-gray-200 rounded-lg shadow-md dark:bg-gray-800 dark:border-gray-700">
            <img className="rounded-t-lg w-full" src={imgSrc} alt="Product-img" />
            <div className="p-5">
                <h5 className="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white">Backpack</h5>
                <p className="mb-3 font-normal text-gray-700 dark:text-gray-400">Here are the biggest enterprise technology acquisitions of 2021 so far, in reverse chronological order.</p>
                <button type='submit' className="inline-flex items-center text-background px-3 py-2 rounded bg-secondary ml-2 hover:bg-primary transition duration-300">
                    Read more
                </button>
            </div>
        </div>

    )
}

export default ItemCard