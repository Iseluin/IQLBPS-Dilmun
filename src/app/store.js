
import {
  configureStore
} from '@reduxjs/toolkit';
import userReducer from '../features/user/userSlice';
import itemReducer from '../features/slices/item'

export const store = configureStore({
  reducer: {
    user: userReducer,
    item: itemReducer
  },
});
