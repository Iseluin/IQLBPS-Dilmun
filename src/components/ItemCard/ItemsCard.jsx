function ItemsCard({ item }) {
  return (
    <div
      data-testid="item-card"
      className="xl:w-80 lg:w-72 md:w-80 w-full bg-background rounded-lg hover:shadow-secondary shadow-gray-400 shadow-lg hover:shadow-2xl duration-300"
    >
      <figure className="relative max-w-sm transition-all cursor-pointer">
        <a href="/">
          <img
            className="rounded-t-lg overflow-hidden object-cover mb-3 lg:h-80 h-80 shadow-md md:w-80 xl:w-full lg:w-72 w-full"
            src={item.file}
            alt="product"
          />
          <figcaption className="absolute top-3 right-1 px-4 text-lg">
            {item.type === 'Donated' ? (
              <span className="inline-block bg-background bg-opacity-80 rounded-xl px-3 py-1 text-xl font-bold text-primary mr-2 mb-2">
                {item.type}
              </span>
            ) : (
              <span className="inline-block bg-background bg-opacity-80 rounded-xl px-3 py-1 text-xl font-bold text-primary mr-2 mb-2">
                {item.price}
              </span>
            )}
          </figcaption>
        </a>
      </figure>
      <div className="px-5 pb-5">
        {item.title.length > 15 ? (
          <h5 className="text-3xl pb-2 font-extrabold text-primary">
            {item.title.slice(0, 1).toUpperCase()}
            {item.title.slice(1, 8)} ...
          </h5>
        ) : (
          <h5 className="text-3xl pb-2 font-extrabold text-primary">
            {item.title.slice(0, 1).toUpperCase()}
            {item.title.slice(1)}
          </h5>
        )}
        <span className="text-xl py-1 font-semibold text-primary">
          {item.type}
        </span>
        {item.description.length > 20 ? (
          <h5 className="text-base py-1 font-base text-primary">
            {item.description.slice(0, 1).toUpperCase()}
            {item.description.slice(1, 30)}...
          </h5>
        ) : (
          <h5 className="text-base py-1 font-base text-primary">
            {item.description.slice(0, 1).toUpperCase()}
            {item.description.slice(1)}
          </h5>
        )}
        <div className="">
          {typeof item.category === 'object' ? (
            item.category.slice(0, 3).map((cat) => {
              return (
                <span
                  key={cat}
                  className="text-sm inline-block mr-1 font-semibold text-primary my-2 bg-tertiary py-2 px-4 bg-opacity-90 rounded-lg"
                >
                  {cat}
                </span>
              );
            })
          ) : (
            <span className="text-sm inline-block mr-1 font-semibold text-primary my-2 bg-tertiary py-2 px-4 bg-opacity-90 rounded-lg">
              {item.category}
            </span>
          )}
          <h5 className="text-base py-1 font-base text-primary">
            {item.createdAt}
          </h5>
        </div>
        <button
          type="button"
          className="w-full my-2 duration-300 text-background bg-secondary hover:bg-primary focus:ring-1 focus:outline-none focus:ring-tertiary font-medium rounded-lg text-md px-5 py-2.5 text-center"
        >
          <a href={`mailto: ${item.owner?.email}`}>Email Owner</a>
        </button>
      </div>
    </div>
  );
}
export default ItemsCard;
