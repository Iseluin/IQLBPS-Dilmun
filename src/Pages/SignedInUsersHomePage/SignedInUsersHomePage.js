import React from 'react'
import { useSelector } from 'react-redux'
import { Link } from 'react-router-dom'
import { BsFillCartFill } from 'react-icons/bs'
import { IoIosAddCircle } from 'react-icons/io'
import { user } from '../../features/slices/user'
import { ADD_ITEM_ROUTE, PRODUCT_ROUTE } from '../../route'
import ItemCard from './ItemCard'



const SignedInUsersHomePage = () => {
    const userData = useSelector(user);

    return (
        <div data-testid='SignedInUseresHomePage' className='bg-background text-primary flex flex-col  items-center justify-between'>
            <div className=' bg-home-page-for-signed-in-users flex flex-col justify-center bg-cover bg-no-repeat bg-center w-full h-[400px] text-white shadow-lg'>
                <div className='w-full h-full flex flex-col justify-center p-6 backdrop-blur-sm bg-primary/30'>
                    <h1 className='text-2xl md:text-6xl font-bold text-center text-shadow'>Welcome back, </h1>
                    <h1 className='text-2xl md:text-6xl font-bold text-center mb-2'> {userData.firstName} {userData.lastName}</h1>
                    <h3 className='md:text-3xl font-bold text-center'> Dilmun is your best choice to buy or sell anything</h3>
                    <div className='sm:flex justify-center gap-2 mt-2'>
                        <span className=' flex items-center justify-center gap-2 font-bold md:text-2xl'>Can&apos;t find what you need? Check our list<Link to={PRODUCT_ROUTE}><BsFillCartFill size={40} className='hover:text-primary transition duration-300' /></Link> </span>
                        <span className=' flex items-center justify-center gap-2 font-bold md:text-2xl'>or add an item <Link to={ADD_ITEM_ROUTE}><IoIosAddCircle size={40} className='hover:text-primary transition duration-300' /></Link> </span>
                    </div>


                </div>
            </div>
            <div className=''>
                <h1 className='text-5xl font-bold my-7 text-center'>Most Recent Items</h1>
                {/* Create a new issue to adjust card height when ItemCard is finished */}
                <div className='w-full px-5 lg:px-0 overflow-x-scroll scrollbar--none flex flex-col lg:flex-row gap-4'>
                    <ItemCard />
                    <ItemCard />
                    <ItemCard />
                    <ItemCard />
                    <ItemCard />
                </div>
            </div>



        </div>
    )
}

export default SignedInUsersHomePage