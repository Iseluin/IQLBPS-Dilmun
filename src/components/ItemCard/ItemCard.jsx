import { useState, useEffect } from 'react';
import { getItemList, itemList } from '../../features/slices/item';
import { useSelector, useDispatch } from '../../app/store';

function ItemCard() {
  const itemData = useSelector(itemList);
  const [items, setItems] = useState([]);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getItemList(itemData));
    setItems(itemData);
  }, []);
  return (
    <div data-testid='item-card' className="w-full grid gap-4 lg:grid-cols-3 md:grid-cols-2 grid-cols-1 content-center items-center justify-items-center bg-background py-6 my-2">
      {items.map((item) => {
        return (
          <div className="lg:w-80 md:w-64 w-96 bg-background m-7 rounded-lg hover:shadow-secondary shadow-gray-400 shadow-lg hover:shadow-2xl duration-300">
            <figure className="relative max-w-sm transition-all cursor-pointer">
              <a href="/">
                <img
                  className="rounded-t-lg overflow-hidden mb-3 lg:h-80 md:h-64 h-96 shadow-md md:w-64 lg:w-80 w-96"
                  src={item.file}
                  alt="product"
                />
                <figcaption className="absolute top-3 right-1 px-4 text-lg">
                  {item.type === 'Donated' ? (
                    <span className="inline-block bg-background bg-opacity-80 rounded-xl px-3 py-1 text-xl font-bold text-primary mr-2 mb-2">
                      {item.type}
                    </span>
                  ) : (
                    <span className="inline-block bg-background bg-opacity-80 rounded-xl px-3 py-1 text-xl font-bold text-primary mr-2 mb-2">
                      {item.price}
                    </span>
                  )}
                </figcaption>
              </a>
            </figure>
            <div className="px-5 pb-5">
              <h5 className="text-3xl pb-2 font-extrabold text-primary">
                {item.title.slice(0, 1).toUpperCase()}
                {item.title.slice(1)}
              </h5>
              <span className="text-xl py-1 font-semibold text-primary">
                {item.type}
              </span>
              {item.description.length > 20 ? (
                <h5 className="text-base py-1 font-base text-primary">
                  {item.description.slice(0, 1).toUpperCase()}
                  {item.description.slice(1, 30)}...
                </h5>
              ) : (
                <h5 className="text-base py-1 font-base text-primary">
                  {item.description.slice(0, 1).toUpperCase()}
                  {item.description.slice(1)}
                </h5>
              )}
              <div className="">
                {typeof item.category === 'object' ? (
                  item.category.map((cat) => {
                    return (
                      <span className="text-sm inline-block mr-1 font-semibold text-primary my-2 bg-tertiary py-2 px-4 bg-opacity-90 rounded-lg">
                        {cat}
                      </span>
                    );
                  })
                ) : (
                  <span className="text-sm inline-block mr-1 font-semibold text-primary my-2 bg-tertiary py-2 px-4 bg-opacity-90 rounded-lg">
                    {item.category}
                  </span>
                )}
                <h5 className="text-base py-1 font-base text-primary">
                  {item.createdAt.toString()}
                </h5>
              </div>
              <button
                type="button"
                className="w-full my-2 duration-300 text-background bg-secondary hover:bg-primary focus:ring-1 focus:outline-none focus:ring-tertiary font-medium rounded-lg text-md px-5 py-2.5 text-center"
              >
                Email Owner
              </button>
            </div>
          </div>
        );
      })}
    </div>
  );
}
export default ItemCard;
