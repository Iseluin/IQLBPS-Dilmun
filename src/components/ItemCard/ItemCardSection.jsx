import { useState, useEffect } from 'react';
import { useSelector, useDispatch } from '../../app/store';
import { getItemList, itemList } from '../../features/slices/item';
import ItemCard from './ItemCard';

export default function ItemCardSection() {
  const itemData = useSelector(itemList);
  const [items, setItems] = useState([]);
  const dispatch = useDispatch();

  const func = async () => {
    await dispatch(getItemList(itemData));
    setItems(itemData.docs.map((d) => {return ({...d.doc(), id: d.id} )}));
  };
  useEffect(() => {
    func();
  }, [items]);

  return (
    <div
      data-testid="item-card"
      className="w-full grid gap-4 lg:grid-cols-3 md:grid-cols-2 grid-cols-1 content-center items-center justify-items-center bg-background py-6 my-2"
    >
      <div>
        {items.map((item) => {
          return <ItemCard item={item} />;
        })}
      </div>
    </div>
  );
}
