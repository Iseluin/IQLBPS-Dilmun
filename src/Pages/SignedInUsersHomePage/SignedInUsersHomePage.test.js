import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../../app/store';
import ItemCard from "./ItemCard"
import SignedInUsersHomePage from "./SignedInUsersHomePage";

test('SignedInUsersHomePage renders without crashing', () => {
  render(
    <Provider store={store}>
      <Router>
        <SignedInUsersHomePage>
          <ItemCard />
        </SignedInUsersHomePage>
      </Router>
    </Provider>
  );
  expect(screen.getByTestId('SignedInUseresHomePage')).toMatchSnapshot();
});

