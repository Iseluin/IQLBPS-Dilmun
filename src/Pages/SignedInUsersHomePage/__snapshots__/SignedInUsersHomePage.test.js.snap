// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignedInUsersHomePage renders without crashing 1`] = `
<div
  class="bg-background text-primary flex flex-col  items-center justify-between"
  data-testid="SignedInUseresHomePage"
>
  <div
    class=" bg-home-page-for-signed-in-users flex flex-col justify-center bg-cover bg-no-repeat bg-center w-full h-[400px] text-white shadow-lg"
  >
    <div
      class="w-full h-full flex flex-col justify-center p-6 backdrop-blur-sm bg-primary/30"
    >
      <h1
        class="text-2xl md:text-6xl font-bold text-center text-shadow"
      >
        signed_in_home_page.welcome
        ,
      </h1>
      <h1
        class="text-2xl md:text-6xl font-bold text-center mb-2"
      >
         
         
      </h1>
      <h3
        class="md:text-3xl font-bold text-center"
      >
         
        signed_in_home_page.best_choice
      </h3>
      <div
        class="sm:flex justify-center gap-2 mt-2"
      >
        <span
          class=" flex items-center justify-center gap-2 font-bold md:text-2xl"
        >
          signed_in_home_page.cant_find
          <a
            href="/products"
          >
            <svg
              class="hover:text-primary transition duration-300"
              fill="currentColor"
              height="40"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 1.5A.5.5 0 0 1 .5 1H2a.5.5 0 0 1 .485.379L2.89 3H14.5a.5.5 0 0 1 .491.592l-1.5 8A.5.5 0 0 1 13 12H4a.5.5 0 0 1-.491-.408L2.01 3.607 1.61 2H.5a.5.5 0 0 1-.5-.5zM5 12a2 2 0 1 0 0 4 2 2 0 0 0 0-4zm7 0a2 2 0 1 0 0 4 2 2 0 0 0 0-4zm-7 1a1 1 0 1 1 0 2 1 1 0 0 1 0-2zm7 0a1 1 0 1 1 0 2 1 1 0 0 1 0-2z"
              />
            </svg>
          </a>
           
        </span>
        <span
          class=" flex items-center justify-center gap-2 font-bold md:text-2xl"
        >
          signed_in_home_page.add_item
          <a
            href="/add-item"
          >
            <svg
              class="hover:text-primary transition duration-300"
              fill="currentColor"
              height="40"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48zm90.5 224H272v74.5c0 8.8-7.2 16-16 16-4.4 0-8.4-1.8-11.3-4.7-2.9-2.9-4.7-6.9-4.7-11.3V272h-74.5c-4.4 0-8.4-1.8-11.3-4.7-2.9-2.9-4.7-6.9-4.7-11.3 0-8.8 7.2-16 16-16H240v-74.5c0-8.8 7.2-16 16-16s16 7.2 16 16V240h74.5c8.8 0 16 7.2 16 16s-7.2 16-16 16z"
              />
            </svg>
          </a>
           
        </span>
      </div>
    </div>
  </div>
  <div
    class=""
  >
    <h1
      class="text-5xl font-bold my-7 text-center"
    >
      signed_in_home_page.most_recent
    </h1>
    <div
      class="w-full px-5 lg:px-0 overflow-x-scroll scrollbar--none flex flex-col lg:flex-row gap-4"
    >
      <div
        class="min-w-fit lg:animate-scroll h-full bg-white border border-gray-200 rounded-lg shadow-md dark:bg-gray-800 dark:border-gray-700"
      >
        <img
          alt=""
          class="rounded-t-lg h-1/2"
          src="https://placeimg.com/800/480/tech/grayscale"
        />
        <div
          class="p-5"
        >
          <h5
            class="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white"
          >
            Backpack
          </h5>
          <p
            class="mb-3 font-normal text-gray-700 dark:text-gray-400"
          >
            Here are the biggest enterprise technology acquisitions of 2021 so far, in reverse chronological order.
          </p>
          <button
            class="inline-flex items-center px-3 py-2 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
            type="submit"
          >
            Read more
            <svg
              aria-hidden="true"
              class="w-4 h-4 ml-2 -mr-1"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="min-w-fit lg:animate-scroll h-full bg-white border border-gray-200 rounded-lg shadow-md dark:bg-gray-800 dark:border-gray-700"
      >
        <img
          alt=""
          class="rounded-t-lg h-1/2"
          src="https://placeimg.com/800/480/tech/grayscale"
        />
        <div
          class="p-5"
        >
          <h5
            class="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white"
          >
            Backpack
          </h5>
          <p
            class="mb-3 font-normal text-gray-700 dark:text-gray-400"
          >
            Here are the biggest enterprise technology acquisitions of 2021 so far, in reverse chronological order.
          </p>
          <button
            class="inline-flex items-center px-3 py-2 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
            type="submit"
          >
            Read more
            <svg
              aria-hidden="true"
              class="w-4 h-4 ml-2 -mr-1"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="min-w-fit lg:animate-scroll h-full bg-white border border-gray-200 rounded-lg shadow-md dark:bg-gray-800 dark:border-gray-700"
      >
        <img
          alt=""
          class="rounded-t-lg h-1/2"
          src="https://placeimg.com/800/480/tech/grayscale"
        />
        <div
          class="p-5"
        >
          <h5
            class="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white"
          >
            Backpack
          </h5>
          <p
            class="mb-3 font-normal text-gray-700 dark:text-gray-400"
          >
            Here are the biggest enterprise technology acquisitions of 2021 so far, in reverse chronological order.
          </p>
          <button
            class="inline-flex items-center px-3 py-2 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
            type="submit"
          >
            Read more
            <svg
              aria-hidden="true"
              class="w-4 h-4 ml-2 -mr-1"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="min-w-fit lg:animate-scroll h-full bg-white border border-gray-200 rounded-lg shadow-md dark:bg-gray-800 dark:border-gray-700"
      >
        <img
          alt=""
          class="rounded-t-lg h-1/2"
          src="https://placeimg.com/800/480/tech/grayscale"
        />
        <div
          class="p-5"
        >
          <h5
            class="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white"
          >
            Backpack
          </h5>
          <p
            class="mb-3 font-normal text-gray-700 dark:text-gray-400"
          >
            Here are the biggest enterprise technology acquisitions of 2021 so far, in reverse chronological order.
          </p>
          <button
            class="inline-flex items-center px-3 py-2 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
            type="submit"
          >
            Read more
            <svg
              aria-hidden="true"
              class="w-4 h-4 ml-2 -mr-1"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="min-w-fit lg:animate-scroll h-full bg-white border border-gray-200 rounded-lg shadow-md dark:bg-gray-800 dark:border-gray-700"
      >
        <img
          alt=""
          class="rounded-t-lg h-1/2"
          src="https://placeimg.com/800/480/tech/grayscale"
        />
        <div
          class="p-5"
        >
          <h5
            class="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white"
          >
            Backpack
          </h5>
          <p
            class="mb-3 font-normal text-gray-700 dark:text-gray-400"
          >
            Here are the biggest enterprise technology acquisitions of 2021 so far, in reverse chronological order.
          </p>
          <button
            class="inline-flex items-center px-3 py-2 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
            type="submit"
          >
            Read more
            <svg
              aria-hidden="true"
              class="w-4 h-4 ml-2 -mr-1"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
